src/browser/api_protocol/api_handlers/browser_view.ts

```diff
@@ -0,0 +1,47 @@
+import * as apiProtocolBase from './api_protocol_base';
+import { ActionSpecMap } from '../shapes';
+import { Channel } from '../../api/channel';
+import { Identity, APIMessage, ProviderIdentity } from '../../../shapes';
+import { AckFunc, NackFunc } from '../transport_strategy/ack';
+import * as browser_view from '../../api/browser_view';
+import { getBrowserViewByIdentity } from '../../core_state';
+
+const successAck = {
+    success: true
+};
+
+async function create (identity: Identity, message: APIMessage) {
+    const { payload } = message;
+    await browser_view.create(payload);
+    return successAck;
+}
+async function attach (identity: Identity, message: APIMessage) {
+    const { payload } = message;
+    const { uuid, name, target } = payload;
+    const view = getBrowserViewByIdentity({uuid, name});
+    await browser_view.attach(view, target);
+    return successAck;
```

Comment:
need to try/catch this as well

Reply:
also caught by middleware because it's an async function.